File: Data Structures & Algorithms/three-integer-sum/submission-0.cs
public class Solution {
    public List<List<int>> ThreeSum(int[] nums)
    {
        Array.Sort(nums);
        List<List<int>> triplets = new List<List<int>>();

        for (int i = 0; i < nums.Length; i++)
        {
            if (i > 0 && nums[i - 1] == nums[i]) continue;

            int l = i + 1;
            int r = nums.Length - 1;

            while (l < r)
            {
                var currSum = nums[l] + nums[r] + nums[i];

                if (currSum == 0)
                {
                    triplets.Add(new List<int> { nums[i], nums[l], nums[r] });
                    l++;
                    while (nums[l] == nums[l - 1] && l < r)
                    {
                        l++;
                    }
                }

                else if (currSum > 0)
                {
                    r--;
                }
                else
                {
                    l++;
                }
            }
        }

        return triplets;
    }

}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, r = 0, 0

        currSum = nums[0]
        minLength = len(nums) + 1

        while l <= r and r < len(nums):
            while currSum < target:
                r += 1
                if r >= len(nums):
                    break
                
                currSum += nums[r]
            
            
            while r < len(nums) and l <= r and currSum >= target:
                minLength = min(minLength, r - l + 1)
                currSum -= nums[l]
                l += 1
            

        if minLength == len(nums) + 1:
            return 0
        else:
            return minLength 


            

                

